### PR TITLE
Task 1. (Cargo) operator== & Task 2. (Cargo) Constructor

### LINKED ISSUE #2: Task 1. (Cargo) operator==
W klasie Cargo napisz operator porównania (operator==), który będzie sprawdzał, czy towary są takie same.

### LINKED ISSUE #3: Task 2. (Cargo) Constructor
Do klasy Cargo dopisz gettery oraz odpowiedni konstruktor, który wypełni wszystkie pola tej klasy.

### PR DESCRIPTION
**Pull request by kasiasieradzka**
resolve #2 resolve #3

### CODE REVIEW

**Comment by Kasperice** [comment]
Looks good for me. It will be easier for me to merge that after #11 PR which implements Ship and Cargo from classes.

**Review by Kasperice — APPROVED** [review]

File: shm/cargo.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+Cargo();
+Cargo(size_t amount, std::string name, size_t base_price)
+    : amount_(amount), name_(name), base_price_(base_price)
+{}
+~Cargo() {;} 
+
+std::string getName() const { return name_; }
+size_t getAmount() const { return amount_; }
+size_t getBase_price() const { return base_price_; }
```
Comment (Kasperice) [inline shm/cargo.cpp]: Let's do not mix two naming conventions
```suggestion
size_t getBasePrice() const { return base_price_; }
```

**Review by Kasperice — CHANGES_REQUESTED** [review]

File: shm/cargo.hpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <string>
+
+class Cargo {
+  private: 
+     std::string name_;
+     size_t amount_;
+     size_t base_price_;
```
Comment (Kasperice) [inline shm/cargo.hpp]: We decided on Discord to use camelCase for functions and variables

```suggestion
     size_t basePrice_;
```

Comment (Kasperice) [reply shm/cargo.hpp]: Please change it everywhere

File: shm/cargo.hpp [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  public: 
+     Cargo();
```
Comment (Kasperice) [inline shm/cargo.hpp:12]: As I see sometimes your indentation is 4 sometimes 5, change it to 4.

**Review by Kasperice — CHANGES_REQUESTED** [review]

File: shm/cargo.hpp [hunk 4]
```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <string>
+
+class Cargo {
+  private: 
+     std::string name_;
+     size_t amount_;
```
Comment (g00fer) [inline shm/cargo.hpp]: It is a wrong use of size_t. size_t is  for indexing arrays, vectors etc. https://en.cppreference.com/w/cpp/types/size_t
Ideally type of amount should be the measurement units.
so ideally: Kilograms amount_;   Then you can do: cargo1.amount_ + cargo2.amount_  But you cannot do cargo2.amount_ + cargo1.base_price_

File: shm/cargo.hpp [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+     size_t basePrice_;
```
Comment (g00fer) [inline shm/cargo.hpp]: Use normal type here (int, unisigned int , uint32_t or typedef (e.g Gold))

Comment (Kasperice) [reply shm/cargo.hpp]: For amount and basePrice (and similar ones) i used uint8_t it should be enough for now

Comment (Kasperice) [reply shm/cargo.hpp]: @kasiasieradzka sorry for confusion, my usage of uint8_ was not the best idea. As @g00fer said uint32_t will be better. If you want you can change that, if not I will change it during merge

File: shm/cargo.cpp [hunk 6]
```diff
@@ -0,0 +1,15 @@
+#include "cargo.hpp"
+
+Cargo();
```
Comment (Pawellos) [inline shm/cargo.cpp]: I think it will not complie, there should be scope, like this:
Cargo::Cargo();

File: shm/cargo.cpp [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+Cargo(uint8_t amount, std::string name, uint8_t base_price)
+    : amount_(amount), name_(name), base_price_(base_price)
+{}
+~Cargo() {;} 
+
+std::string getName() const { return name_; }
+uint8_t getAmount() const { return amount_; }
+uint8_t getBase_price() const { return base_price_; }
```
Comment (Pawellos) [inline shm/cargo.cpp]: every where with methods should be :

std::string Cargo::getName() const { return name_; }
uint8_t Cargo::getAmount() const { return amount_; }
uint8_t Cargo::getBase_price() const { return base_price_; }

in my opinion

**Review by Pawellos — CHANGES_REQUESTED** [review]

**Review by g00fer — CHANGES_REQUESTED** [review]
Please make sure it is compililing.

File: shm/cargo.cpp (hunk 6 above)
Comment (kasiasieradzka) [reply shm/cargo.cpp]: But I think we use Cargo::Cargo(); when we are outside of class, inside we don't need it

**Comment by kasiasieradzka** [comment]
@g00fer can you tell me how compiling it on github?

File: shm/cargo.cpp (hunk 6 above)
Comment (Kasperice) [reply shm/cargo.cpp]: Everywhere I saw for constructor is ClassName::ClassName(). As I remember even VSC underline it if I did not use ClassName::ClassName(), but someone more experienced could clarify

File: shm/cargo.hpp (hunk 5 above)
Comment (kasiasieradzka) [reply shm/cargo.hpp]: @Kasperice, it fixed

Comment (Kasperice) [reply shm/cargo.hpp]: thanks :)

File: shm/cargo.cpp (hunk 6 above)
Comment (kasiasieradzka) [reply shm/cargo.cpp]: Yes, I mixed up everything... Give me few minutes to fix it

**Review by Kasperice — APPROVED** [review]

**Pull request merged**